actors/builtin/miner/miner_state.go

```diff
@@ -156,7 +159,12 @@ func (st *State) HasSectorNo(store adt.Store, sectorNo abi.SectorNumber) (bool,
 	var info SectorOnChainInfo
 	found, err := sectors.Get(uint64(sectorNo), &info)
 	if err != nil {
-		return false, errors.Wrapf(err, "failed to get sector %v", sectorNo)
+		var enf *amt.ErrNotFound
+		if xerrors.As(err, &enf) {
+			return false, nil
+		}
```

Comment:
This looks messed up. `Array.Get` already returns a boolean indicating found, checking for the same error type. If there's something wrong with that check, can we resolve it lower down?

Reply:
Yeah, it wasnt happy. Let's see if I can sink it deeper, could have been the wrong solution to the problem